src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/DecryptorPalWindows.Decrypt.cs

```diff
@@ -17,8 +17,60 @@ namespace Internal.Cryptography.Pal.Windows
 {
     internal sealed partial class DecryptorPalWindows : DecryptorPal
     {
-        public sealed override ContentInfo TryDecrypt(RecipientInfo recipientInfo, X509Certificate2 cert, X509Certificate2Collection originatorCerts, X509Certificate2Collection extraStore, out Exception exception)
+        public unsafe sealed override ContentInfo TryDecrypt(
+            RecipientInfo recipientInfo,
+            X509Certificate2 cert,
+            AsymmetricAlgorithm privateKey,
+            X509Certificate2Collection originatorCerts,
+            X509Certificate2Collection extraStore,
+            out Exception exception)
         {
+            Debug.Assert((cert != null) ^ (privateKey != null));
+
+            if (privateKey != null)
+            {
+                RSA key = privateKey as RSA;
+
+                if (key == null)
+                {
+                    exception = new CryptographicException(SR.Cryptography_Cms_Decrypting_RequiresRSAPrivateKey);
+                    return null;
+                }
+
+                ContentInfo contentInfo = _hCryptMsg.GetContentInfo();
+                byte[] cek = AnyOS.ManagedPkcsPal.ManagedKeyTransPal.DecryptCekCore(
+                                cert,
```

Comment:
The arguments here seem a bit overly indented.

Reply:
I indented so it's more that `=` will leave at `+4`

Reply:
Please don't try any fancy alignment.  The style in the rest of this project would be

```C#
byte[] cek = AnyOS.ManagedPkcsPal.ManagedKeyTransPal.DecryptCekCore(
    cert,
    ...);
```

Renaming the variable shouldn't require re-indenting the subsequent lines.